conbench/cli.py

```diff
@@ -81,6 +86,61 @@ def _to_cli_name(name):
     return f"--{name.replace('_', '-')}"
 
 
+def log_timing_report_across_results(results):
+    # Find tags that are common among all results. That makes sense only when
+    # there is more than one result.
+    if len(results) == 1:
+        common_tag_keys = []
+    else:
+        common_tags = set.intersection(*tuple(set(r["tags"].items()) for r in results))
+        common_tag_keys = [t[0] for t in common_tags]
+        log.info("common tag keys across results: %s", common_tag_keys)
+
+    # For each result, assemble a human-readable string which represents the
+    # case/scenario from the tags. Also assemble a human-readable string that
+    # reports the timing info for this scenario.
+    tagstrings = []
+    timestrings = []
+
+    for r in results:
+        tags = r["tags"]
+
+        # That represents the case/senario. Note that `tags` has been enriched
+        # with key/value pairs beyond the case -- filter these out again.
+        tagstring = ", ".join(
+            f"{k}: {tags[k]}" for k in sorted(tags.keys()) if k not in common_tag_keys
+        )
```

Comment:
We have similar behavior [in hacks.py](https://github.com/conbench/conbench/blob/main/conbench/hacks.py#L1) (which itself has inspired issues like #400). Would you consider consolidating that code? I like this strategy of filtering out common tags instead of a hardcoded list of keys (since technically tags is a freeform dict) but I understand it will take some thought to apply that strategy to what exists in hacks.py today.

Reply:
> We have similar behavior [in hacks.py](https://github.com/conbench/conbench/blob/main/conbench/hacks.py#L1) (which itself has inspired issues like https://github.com/conbench/conbench/issues/400).

lol, now I understand what this was/is about :-) and thank you for the thorough response on #400 -- I had missed this!

Reply:
>  I like this strategy of filtering out common tags instead of a hardcoded list of keys

Cool, thanks for this thought. Then I suppose I now 'know' (roughly) how to consolidate these two.... :shrug: 

I will see about that.